src/Orleans.Core/Manifest/DiagnosticInfoGrainPropertiesProvider.cs

```diff
@@ -0,0 +1,22 @@
+using System;
+using System.Collections.Generic;
+using Orleans.Runtime;
+using Orleans.Utilities;
+
+namespace Orleans.Metadata
+{
+    internal sealed class DiagnosticInfoGrainPropertiesProvider : IGrainPropertiesProvider, IGrainInterfacePropertiesProvider
+    {
+        public void Populate(Type grainClass, GrainType grainType, Dictionary<string, string> properties)
+        {
+            properties["diag.type"] = RuntimeTypeNameFormatter.Format(grainClass);
+            properties["diag.asm"] = grainClass.Assembly.GetName().Name;
+        }
+
+        public void Populate(Type interfaceType, GrainInterfaceId interfaceId, Dictionary<string, string> properties)
+        {
+            properties["diag.type"] = RuntimeTypeNameFormatter.Format(interfaceType);
+            properties["diag.asm"] = interfaceType.Assembly.GetName().Name;
```

Comment:
really NIT: duplicate strings

Reply:
I figured these diagnostic properties didn't warrant an entry in the WellKnown* classes

Reply:
(because I don't want anyone to rely on them)

Reply:
meh it was mostly to have some private string const in the class to avoid duplicate string in memory, but the compiler is probably clever enough

Reply:
Yeah, it is - it will intern them